lib/block-supports/layout.php

```diff
@@ -233,3 +233,36 @@ function( $matches ) {
 	remove_filter( 'render_block', 'wp_restore_group_inner_container', 10, 2 );
 }
 add_filter( 'render_block', 'gutenberg_restore_group_inner_container', 10, 2 );
+
+
+/**
+ * For themes without theme.json file, make sure
+ * to restore the outer div for the aligned image block
+ * to avoid breaking styles relying on that div.
+ *
+ * @param string $block_content Rendered block content.
+ * @param array  $block         Block object.
+ * @return string Filtered block content.
+ */
+function gutenberg_restore_image_outer_container( $block_content, $block ) {
+	$image_with_align = '/(^\s*<figure\b[^>]*)wp-block-image\s([^"]*((alignleft)|(alignright)|(aligncenter))(\s|")[^>]*>((.|\S|\s)*)<\/figure>)/U';
```

Comment:
We can probably return early if the block isn't an image or doesn't have `theme.json`.

```php
if (
  'core/image' !== $block['blockName'] ||
  WP_Theme_JSON_Resolver::theme_has_support()
) {
  return $block_content;
}
```

P.S. Sorry, I'm also not an expert here.

Reply:
yeah, that's what it does already but since this is just an assignment, I think it's fine. (the variable is also used for the early return)

Reply:
Notes from reviewing the regular expression:

1. Word boundary needed before "wp-block-image", otherwise you would match `<figure class="not-wp-block-image […]`. **Fix:** `\bwp-block-image`.

2. The RE doesn't target the `class` attribute specifically; shouldn't it? Right now I can match `<figure not-class="...`.

3. No parens needed around each string: `(alignleft|alignright|aligncenter)`

4. Many of the remaining groups don't need capturing: we only care about `$match[1]` and `$match[2]`, so the above example could become `(?:alignleft|alignright:aligncenter)`, and same for `(?:\s|")`.

5. `((.|\S|\s)*)` is amusing! `\S|\s` means "I want to match something as well as its opposite", i.e. "everything", which is the same as `.`. So, unless I missed something, the outer group could be `.*`

6. `(\s|")`: I'm not sure what purpose this serves. Oh, maybe it's used as a word boundary after the alignment class? Right, and now I notice that there is missing word boundary at the start of the alignment class. You'll see that right now you're able to (incorrectly) match `zzzzalignleft`. So let's use proper word boundaries on both sides. **Fix:** `\b(?:alignleft|alignright|aligncenter)\b`.

7. (I don't think I'd ever seen the U (ungreedy) flag in use, that's cool)

Based on all of the above except for the fact that we don't target the `class` attribute, the RE could be:

```
/(^\s*<figure\b[^>]*)\bwp-block-image\b([^"]*\b(?:alignleft|alignright|aligncenter)\b[^>]*>.*<\/figure>)/U
```

I've tested this in the venerable [regexr.com](https://regexr.com), but not with live content, so take my review accordingly. :)

Reply:
2- It could target class to be 100% accurate but I didn't think it was mandatory (like group blocks).

> 5- ((.|\S|\s)*) is amusing! \S|\s means "I want to match something as well as its opposite", i.e. "everything", which is the same as .. So, unless I missed something, the outer group could be .*

Indeed, but for some reason in the site I was using `.` was not catching everything (I think related to multiline)

6- `(\s|")` this one targets a space or end of attribute (if it's the last class)

Reply:
> For the first remark, it could also be "wp-block-image, is " a word boundary?

That should work just as well, if you're sure that `wp-block-image` is always the first class.

> Indeed, but for some reason in the site I was using . was not catching everything (I think related to multiline)

Hm, interesting. Take my advice with caution, then. :) But make sure the trick is really necessary.